classes/announcement/Collector.php

```diff
@@ -165,53 +169,39 @@ public function orderBy(?string $sorter, string $direction = self::ORDER_DIR_DES
+
+        $includeSite = in_array($this->includeSite, [static::SITE_AND_CONTEXTS, static::SITE_ONLY]);
+        if (isset($this->contextIds) || $includeSite) {
+            $qb->whereIn(DB::raw('COALESCE(a.assoc_id, 0)'), array_merge($this->contextIds ?? [], $includeSite ? [0] : []));
         }
 
         if (isset($this->typeIds)) {
             $qb->whereIn('a.type_id', $this->typeIds);
         }
 
-        $qb->when($this->isActive, fn ($qb) => $qb->where(function ($qb) {
-            $qb->where('a.date_expire', '>', $this->isActive)
-                ->orWhereNull('a.date_expire');
-        }));
-
-        if ($this->searchPhrase !== null) {
-            $words = explode(' ', $this->searchPhrase);
-            if (count($words)) {
-                $qb->whereIn('a.announcement_id', function ($query) use ($words) {
-                    $query->select('announcement_id')->from($this->dao->settingsTable);
-                    foreach ($words as $word) {
-                        $word = strtolower(addcslashes($word, '%_'));
-                        $query->where(function ($query) use ($word) {
-                            $query->where(function ($query) use ($word) {
-                                $query->where('setting_name', 'title');
-                                $query->where(DB::raw('lower(setting_value)'), 'LIKE', "%{$word}%");
-                            })
-                                ->orWhere(function ($query) use ($word) {
-                                    $query->where('setting_name', 'descriptionShort');
-                                    $query->where(DB::raw('lower(setting_value)'), 'LIKE', "%{$word}%");
-                                })
-                                ->orWhere(function ($query) use ($word) {
-                                    $query->where('setting_name', 'description');
-                                    $query->where(DB::raw('lower(setting_value)'), 'LIKE', "%{$word}%");
-                                });
-                        });
-                    }
-                });
-            }
+        $qb->when(
+            $this->isActive,
+            fn (Builder $qb) => $qb->where(fn (Builder $qb) => $qb->where('a.date_expire', '>', $this->isActive)->orWhereNull('a.date_expire'))
+        );
+
+        $searchPhrase = trim($this->searchPhrase ?? '');
+        if (strlen($searchPhrase)) {
+            $words = PKPString::regexp_split('/\s+/', $searchPhrase);
+            $qb->whereIn('a.announcement_id', function (Builder $q) use ($words) {
+                $q->select('announcement_id')->from($this->dao->settingsTable);
+                foreach ($words as $word) {
+                    $word = addcslashes($word, '%_');
+                    $q->where(
+                        fn (Builder $q) => $q
+                            ->where(fn (Builder $q) => $q->where('setting_name', 'title')->whereRaw('LOWER(setting_value) LIKE ?', ["%{$word}%"]))
```

Comment:
I think here and below `LOWER(?) `should be, no?

Reply:
YES, this was a half-baked update 😂